packages/frontend/app/routes/payments.incoming.$incomingPaymentId.tsx

```diff
@@ -0,0 +1,188 @@
+              </Badge>
+            </p>
+            {new Date(incomingPayment.expiresAt) > new Date() && (
+              <p className='text-sm mb-2'>
+                Expires at {incomingPayment.expiresAt}{' '}
+              </p>
+            )}
+          </div>
+          <div className='md:col-span-2 bg-white rounded-md shadow-md'>
+            <div className='w-full p-4 space-y-3'>
+              <div>
+                <p className='font-medium'>Incoming Payment ID</p>
+                <p className='mt-1'>{incomingPayment.id}</p>
+              </div>
+              <div>
+                <p className='font-medium'>
+                  Wallet Address ID{' '}
+                  <Button
+                    className='mt-1 ml-1'
+                    aria-label='go to wallet address page'
+                    type='button'
+                    size='xs'
+                    to={`/wallet-addresses/${incomingPayment.walletAddressId}`}
+                  >
+                    ↗️
+                  </Button>
+                </p>
+                <p className='mt-1'>{incomingPayment.walletAddressId}</p>
+              </div>
+              <div>
+                <p className='font-medium'>Incoming Amount</p>
+                <p className='mt-1'>
+                  {incomingPayment.incomingAmount ? (
+                    formatAmount(
+                      incomingPayment.incomingAmount.value,
+                      incomingPayment.incomingAmount.assetScale
+                    ) +
+                    ' ' +
+                    incomingPayment.incomingAmount.assetCode
+                  ) : (
+                    <em>None</em>
+                  )}
+                </p>
+              </div>
+              <div>
+                <p className='font-medium'>Received Amount</p>
+                <p className='mt-1'>
+                  {formatAmount(
+                    incomingPayment.receivedAmount.value,
+                    incomingPayment.receivedAmount.assetScale
+                  ) +
+                    ' ' +
+                    incomingPayment.receivedAmount.assetCode}
+                </p>
+              </div>
+              <div>
+                {incomingPayment.metadata ? (
+                  <>
+                    <button
+                      className='font-medium mb-1 cursor-pointer'
+                      aria-label='toggle metadata visibility'
+                      onClick={() => setShowMetadata(!showMetadata)}
+                    >
+                      {showMetadata ? '▼' : '►'} Metadata
+                    </button>
+                    {showMetadata && incomingPayment.metadata && (
+                      <pre
+                        className='mt-1 text-sm'
+                        dangerouslySetInnerHTML={{
+                          __html: prettify(incomingPayment.metadata)
+                        }}
+                      />
+                    )}
+                  </>
```

Comment:
Talked this over with Radu but would welcome another opinion on this. using `dangerouslySetInnerHTML` with user provide data (metadata) is vulnerable to XSS. However, our prettify function sanitizes html chars `<`, `>`, `&` so a metadata key or value like `<script>alert('here's a message from a bad actor with a link to some shady site')</script>` wont work. 

I tested a bunch of things myself but more attempts to disprove that this isnt safe are welcome. Can also simply not use prettify the metadata in which case we dont need to use `dangerouslySetInnerHTML` at all.

Reply:
I think this is ok given we sanitize <>